src/test/java/com/cloudant/tests/DatabaseTest.java

```diff
@@ -71,6 +72,22 @@ public void permissions() {
 
     }
 
+    /**
+     * Test that when called against a DB that is not a Cloudant service
+     * an UnsupportedOperationException is thrown
+     */
+    @Test
+    public void testPermissionsException() {
+        try {
+            Map<String, EnumSet<Permissions>> userPerms = db.getPermissions();
+            //ignore the test in cases where we don't get an exception
+            Assume.assumeTrue(false);

```

Comment:
I always find the logic for `Assume` confusing - not that it's your fault. Can we have a message here like "skipping test as it looks like the DB is a Cloudant service"